File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter04/DisposingSponsor/MathLibrary/DisposingSponsor.vb
Imports System.Runtime.Remoting.Lifetime

Public Class DisposingSponsor
   Implements ISponsor

   Private mSponsoredObject As IDisposable

   Public Sub New(sponsoredObject As IDisposable)
      mSponsoredObject = sponsoredObject
   End Sub

   Public Function Renewal(lease As ILease) As TimeSpan _
      Implements ISponsor.Renewal

      mSponsoredObject.Dispose()
      Return TimeSpan.FromSeconds(0)
   End Function
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter04/DisposingSponsor/MathLibrary/DisposingSponsor.cs
using System;
using System.Runtime.Remoting.Lifetime;

namespace MathLibrary
{
	class DisposingSponsor : ISponsor
	{
      private IDisposable mSponsoredObject;

		public DisposingSponsor(IDisposable sponsoredObject)
		{
			mSponsoredObject = sponsoredObject;
		}

      public TimeSpan Renewal(ILease leaseInfo)
      {
         mSponsoredObject.Dispose();
         return TimeSpan.FromSeconds(0);
      }
	}
}
